src/com/pentalog/pentastagiu/homework2/HomeworkApp.java

```diff
@@ -1,7 +1,184 @@
 package com.pentalog.pentastagiu.homework2;
 
+import com.pentalog.pentastagiu.homework2.bankaccount.BankAccount;
+import com.pentalog.pentastagiu.homework2.bankaccount.impl.BankAccountImpl;
+import com.pentalog.pentastagiu.homework2.boardproblem.Board;
+import com.pentalog.pentastagiu.homework2.boardproblem.Message;
+import com.pentalog.pentastagiu.homework2.boardproblem.User;
+import com.pentalog.pentastagiu.homework2.consolecommands.ConsoleBankAccountCommands;
+import com.pentalog.pentastagiu.homework2.exceptions.InsufficientMoneyException;
+import com.pentalog.pentastagiu.homework2.exceptions.NegativeDepositAmountException;
+import com.pentalog.pentastagiu.homework2.shop.Basket;
+import com.pentalog.pentastagiu.homework2.shop.Customer;
+import com.pentalog.pentastagiu.homework2.shop.Product;
+import com.pentalog.pentastagiu.homework2.shop.customermembership.MembershipType;
+import com.pentalog.pentastagiu.homework2.shop.impl.Book;
+import com.pentalog.pentastagiu.homework2.shop.impl.Candy;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import java.util.function.Supplier;
+
 public class HomeworkApp {
+
+    private static Scanner scanner = new Scanner(System.in);
+
     public static void main(String[] args) {
```

Comment:
in general ca buna practica se sugereaza sa nu ai metode foarte mari pentru ca sunt greu de urmarit, daca sunt cateva linii de cod care pot fi grupate pot fi extrase intr-o metoda cu nume sugestiv si astfel e mai cursiv de citit codul
e bine ca ai extras logica pt fiecare problema intr-o metoda separata, ar mai fi mers sa mai refactorizezei putin (de ex sa ai metoda de setupUsersForBoard in care sa creezi lista de user si sa ii inregistrezi)